packages/clerk-js/src/ui/customizables/parseVariables.ts

```diff
@@ -12,23 +12,31 @@ import {
 
 export const createColorScales = (theme: Theme) => {
   const variables = theme.variables || {};
+
+  //values dependent on other values
+  const textSecondary = toHSLA(variables.colorTextSecondary) || colors.makeTransparent(variables.colorText, 0.15);
+  const textTertiary = toHSLA(variables.colorTextTertiary) || colors.makeTransparent(textSecondary, 0.4);
+
   return removeUndefinedProps({
     ...colorOptionToHslaLightnessScale(variables.colorPrimary, 'primary'),
     ...colorOptionToHslaAlphaScale(variables.colorPrimary, 'primaryAlpha'),
+    ...colorOptionToHslaLightnessScale(variables.colorSecondary, 'secondary'),
     ...colorOptionToHslaLightnessScale(variables.colorDanger, 'danger'),
     ...colorOptionToHslaAlphaScale(variables.colorDanger, 'dangerAlpha'),
     ...colorOptionToHslaLightnessScale(variables.colorSuccess, 'success'),
     ...colorOptionToHslaAlphaScale(variables.colorSuccess, 'successAlpha'),
     ...colorOptionToHslaLightnessScale(variables.colorWarning, 'warning'),
     ...colorOptionToHslaAlphaScale(variables.colorWarning, 'warningAlpha'),
     ...colorOptionToHslaAlphaScale(variables.colorAlphaShade, 'blackAlpha'),
-    colorText: toHSLA(variables.colorText),
-    colorTextOnPrimaryBackground: toHSLA(variables.colorTextOnPrimaryBackground),
-    colorTextSecondary: toHSLA(variables.colorTextSecondary) || colors.makeTransparent(variables.colorText, 0.35),
-    colorInputText: toHSLA(variables.colorInputText),
-    colorBackground: toHSLA(variables.colorBackground),
-    colorInputBackground: toHSLA(variables.colorInputBackground),
-    colorShimmer: toHSLA(variables.colorShimmer),
+    primaryForeground: toHSLA(variables.colorTextOnPrimaryBackground),
+    secondaryForeground: toHSLA(variables.colorTextOnSecondaryBackground),
+    text: toHSLA(variables.colorText),
+    textSecondary: textSecondary,
+    textTertiary: textTertiary,
+    inputForeground: toHSLA(variables.colorInputText),
+    background: toHSLA(variables.colorBackground),
+    input: toHSLA(variables.colorInputBackground),
+    shimmer: toHSLA(variables.colorShimmer),
```

Comment:
This is a bit confusing tbh. I'd expect we keep the same names so that we can be consistent internally and externally with our variables.

Reply:
I'll adjust those when we settle on the new API.

Reply:
Done